src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java

```diff
@@ -263,6 +274,7 @@ private void removeMessagesWithQosZero(boolean needToCheckCurSpoolerSize) {
                 }
             }
         }
+        qos0MessageCheckRequired = false;
```

Comment:
When not removing all messages (For example, to make space for incoming message), it will make sure to set the flag to true after it adds that new message. In case it removes some QOS0 messages and still fails to make space, we are ok with keeping the flag to false (as no need to check for QOS0 messages, as we can't make space even after deleting them)

Reply:
What about case for mqtt disconnect and reconnect?